core/services/relay/delegate/delegate.go

```diff
@@ -0,0 +1,146 @@
+	}
+	return err
+}
+
+// A delegate relayer is healthy if all relayers it manages are ready.
+func (d delegate) Ready() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Ready())
+	}
+	return err
+}
+
+// A delegate relayer is healthy if all relayers it manages are healthy.
+func (d delegate) Healthy() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Healthy())
+	}
+	return err
+}
+
+func (d delegate) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relay.OCR2Provider, error) {
+	// We expect trusted input
+	spec := s.(*job.OffchainReporting2OracleSpec)
+	choice := spec.Relay
+	switch choice {
+	case relay.EVM:
+		var config evmrelay.RelayConfig
+		err := json.Unmarshal(spec.RelayConfig.Bytes(), &config)
+		if err != nil {
+			return nil, err
+		}
+
+		return d.relayers[relay.EVM].NewOCR2Provider(externalJobID, evmrelay.OCR2Spec{
+			ID:             spec.ID,
+			IsBootstrap:    spec.IsBootstrapPeer,
+			ContractID:     spec.ContractID,
+			OCRKeyBundleID: spec.OCRKeyBundleID,
+			TransmitterID:  spec.TransmitterID,
+			ChainID:        config.ChainID.ToInt(),
+		})
+	case relay.Solana:
+		var config solana.RelayConfig
+		err := json.Unmarshal(spec.RelayConfig.Bytes(), &config)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'spec.RelayConfig' unmarshal")
+		}
+
+		programID, err := solanaGo.PublicKeyFromBase58(spec.ContractID)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'solana.PublicKeyFromBase58' for 'spec.ContractID")
+		}
+
+		stateID, err := solanaGo.PublicKeyFromBase58(config.StateID)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'solana.PublicKeyFromBase58' for 'spec.RelayConfig.StateID")
+		}
+		validatorProgramID, err := solanaGo.PublicKeyFromBase58(config.ValidatorProgramID)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'solana.PublicKeyFromBase58' for 'spec.RelayConfig.StateID")
+		}
+		transmissionsID, err := solanaGo.PublicKeyFromBase58(config.TransmissionsID)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'solana.PublicKeyFromBase58' for 'spec.RelayConfig.TransmissionsID")
+		}
+		var transmissionSigner solana.TransmissionSigner
+		if !spec.IsBootstrapPeer {
+			if !spec.TransmitterID.Valid {
+				return nil, errors.New("transmitterID is required for non-bootstrap jobs")
+			}
+			transmissionSigner, err = d.ks.Solana().Get(spec.TransmitterID.String)
+			if err != nil {
+				return nil, err
+			}
+		}
+		return d.relayers[relay.Solana].NewOCR2Provider(externalJobID, solana.OCR2Spec{
+			ID:                 spec.ID,
+			IsBootstrap:        spec.IsBootstrapPeer,
+			NodeEndpointRPC:    config.NodeEndpointRPC,
+			NodeEndpointWS:     config.NodeEndpointWS,
+			ProgramID:          programID,
+			StateID:            stateID,
+			ValidatorProgramID: validatorProgramID,
+			TransmissionsID:    transmissionsID,
+			TransmissionSigner: transmissionSigner,
```

Comment:
The `TransmissionSigner` should not be passed in through the relay OCR2 `spec` which is mostly configuration (serializable data) and the `TransmissionSigner` is a non-serializable object type.

The spec should carry a simple `TransmitterID` data point, which the provider uses to get the `TransmissionSigner` from the relay injected dependencies (could be `d.ks.Solana()` in this case.

IMHO we should be careful and explicit in our design about how functionality is injected vs. how data is shared.

Reply:
Yeah good point. My only hesitancy there is limiting the scope of the relay lib, but I think I can just define a subset of keystore methods as an interface